SDLSecurity/SDLCertificateManager.m

```diff
@@ -62,10 +62,16 @@ - (void)retrieveNewCertificateWithAppId:(NSString *)appId completionHandler:(SDL
 
         if (jsonError != nil || jsonDictionary == nil) {
             SDLSecurityLogE(@"Error parsing network request data");
-            return completionHandler(NO, [NSError errorWithDomain:SDLSecurityErrorDomain code:SDLTLSErrorCodeNoCertificate userInfo:@{NSLocalizedDescriptionKey: @"Network request did not return a certificate"}]);
+            return completionHandler(NO, [NSError errorWithDomain:SDLSecurityErrorDomain code:SDLTLSErrorCodeNoCertificate userInfo:@{NSLocalizedDescriptionKey: @"Network request did not return data in the expected format"}]);
         }
 
-        NSData *certificateData = [[NSData alloc] initWithBase64EncodedData:[jsonDictionary valueForKeyPath:@"data.certificate"] options:0];
+        NSString *certificateKey = @"data.certificate";
+        if (![jsonDictionary valueForKeyPath:certificateKey]) {
```

Comment:
```suggestion
        if ([jsonDictionary valueForKeyPath:certificateKey] == nil) {
```

**Also**
I think you should pull the data out and check if it's `nil` here and use it below. The current way uses `[jsonDictionary valueForKeyPath:certificateKey]` twice. It may be better to do that once and reference the variable instead.

Reply:
Fixed.